logging/telemetryspec/metric.go

```diff
@@ -82,6 +95,37 @@ type AssembleBlockMetrics struct {
 func (m AssembleBlockMetrics) Identifier() Metric {
 	return assembleBlockMetricsIdentifier
 }
+func (m AssembleBlockStats) String() string {
```

Comment:
These are reported as JSON, so the last time I asserted on a telemetry event being reported, I did json.Marshal and then asserted on the JSON output (after deserializing it to a map[string]interface{}).

It seems like you only have this in here for testing... maybe you could move it into your test rather than leaving it here where it might get accidentally used?

Reply:
This is not for testing. Not all redirections of the metrics will report this metrics if this function is not provided.

Reply:
how did it work before then? what is using this now? I guess I am not sure where else this goes, none of the other types in this file have a String() implementation

Reply:
When you configure `Telemetry` with `SendToLog = true`, it will use this function.
If you don't provide this function, the default method will not know how to print the `StateProofStats` field inside `AssembleBlockStats` and skip it.